packages/syft/src/syft/core/tensor/ancestors.py

```diff
@@ -479,12 +495,22 @@ def _private(
         if max_vals.shape != self.child.shape:
             max_vals = lazyrepeatarray(max_vals, self.child.shape)
 
-        self.replace_abstraction_top(
-            tensor_type=_PhiTensor(),
-            child=self.child,
-            min_vals=min_vals,
-            max_vals=max_vals,
-            data_subjects=data_subjects,  # type: ignore
-        )  # type: ignore
+        unique_data_subjects = len(data_subjects.sum())
```

Comment:
Q:  Should we check at the data point level.
for example DataSubjectArray:
<pre><code>
data = np.array([  [1,2] , 
                   [3,4]])
</code></pre>

dsa = [ [ DSA("natsu"), DSA("natsu"],
                DSA("natsu"), DSA("zerek")]]
Since each data point belongs to different entity, do we consider them  as PhiTensor?
The above code would return a Gamma Tensor.

Suppose if the data subjects were 

dsa = [ [ DSA("natsu","zerek"), DSA("zerek","natsu"],
                DSA("natsu"), DSA("zerek")]]

Like in this example , two of the datapoints have more than one data subjects, so this is a GammaTensor?

TLDR: Maybe we should check datasubjectarray to know if any of the data points have more than one datasubjects?